packages/contracts/test/unit/ApeLendingStrategy.t.sol

```diff
@@ -800,4 +831,39 @@ contract ApeLendingStrategyTest is TestWithERC1820Registry {
             abi.encode(currentDebt)
         );
     }
+
+    function testShutdownStrategyForSignificantLoss(uint256 amount, uint256 loss) public {
+        vm.assume(amount > 100 && amount < 10 ** 22 && loss > amount / 4 && amount > loss);
+
+        // Give the required funds to Actor
+        underlying.mint(alice, amount);
+        assertEq(underlying.balanceOf(alice), amount);
+
+        // Allow to vault to take the Actor's assets
+        vm.prank(alice);
+        underlying.increaseAllowance(address(vault), type(uint256).max);
+
+        // Actor deposits funds to the vault
+        vm.prank(alice);
+        vault.deposit(amount);
+
+        assertEq(vault.freeAssets(), amount);
+        assertEq(strategy.depositedBalance(), 0);
+
+        // distribute funds to strategy
+        strategy.callWork();
+
+        assertEq(strategy.depositedBalance(), amount);
+        assertEq(vault.freeAssets(), 0);
+
+        // check if shutdown triggered
+        vm.expectEmit(true, true, true, true);
+        strategy.emitHealthCheckTriggered(2);
+
+        cToken.setLoss(loss);
+        strategy.callWork();
+
+        // check if not lost money returned back to vault after shutdown the strategy
+        assertEq(vault.freeAssets(), amount - loss);
```

Comment:
Also need check that strategy paused and nothing left on strategy balance

Reply:
added